symengine/tests/test_cse.py

```diff
@@ -0,0 +1,17 @@
+from symengine import cse, sqrt, symbols
+
+def test_cse_single():
+    x, y, x0 = symbols("x, y, x0")
+    e = pow(x + y, 2) + sqrt(x + y)
+    substs, reduced = cse([e])
+    assert substs == [(x0, x + y)]
+    assert reduced == [sqrt(x0) + x0**2]
+
+
+def test_multiple_expressions():
+    w, x, y, z, x0 = symbols("w, x, y, z, x0")
+    e1 = (x + y)*z
+    e2 = (x + y)*w
+    substs, reduced = cse([e1, e2])
+    assert substs == [(x0, x + y)]
+    assert reduced == [x0*z, x0*w]
```

Comment:
I only added a few tests here to test the interface. All the tests in C++ are taken from the sympy test suite and there's no point reproducing here.

Reply:
Yes, it's just for the interface / API, not the correctness, which is (and must) be tested in C++.